pkg/pod/status.go

```diff
@@ -300,15 +335,36 @@ func filterResults(results []result.RunResult, specResults []v1.TaskResult) ([]v
 			}
 			taskResults = append(taskResults, taskRunResult)
 			filteredResults = append(filteredResults, r)
+		case result.StepResultType:
+			var stepResult v1.TaskRunResult
+			if neededTypes[r.Key] == v1.ResultsTypeString {
+				stepResult = v1.TaskRunResult{
+					Name:  r.Key,
+					Type:  v1.ResultsTypeString,
+					Value: *v1.NewStructuredValues(r.Value),
```

Comment:
this is not tested, I think maybe it's me that add this code before, will take a look and see if we need it

Reply:
good catch. I will add it now.

Reply:
Actually, it is not needed. The `else` part covers it. Removing it so that it is clean.